telemetry/telemetry_record.py

```diff
@@ -96,9 +156,356 @@ def failed_url(url, output):
+    """Deletes files between trials
+
+    Removes:
+    /home/jamshed/page_load_time/telemetry/benchmark_results.db
+    """
+    commands = [
+        'rm -f '
+        '/home/jamshed/page_load_time/telemetry/temp/benchmark_results.db'
+        ]
+
+    for cmd in commands:
+        p = Popen(cmd, shell=True)
+        p.wait()
+
+def get_min_results():
+    """Converts temp/aggregate.db into results.db, only taking min times
+
+    Format of output:
+    {'http://jamnoise.com':
+        {'cold_time': 314.159,
+         'modified_cold_time': '300.201'},
+         ...
+    }
+    """
+    aggregate_path = \
+            '/home/jamshed/page_load_time/telemetry/temp/aggregate.db'
+    aggregate_data = {}
+    try:
+        aggregate_data = pickle.load(open(aggregate_path, 'rb'))
+    except IOError:
+        raise IOError('Could not read from {0}'.format(aggregate_path))
+
+    min_data = {}
+
+    for url in aggregate_data:
+        url_min_results = {}
+        url_results = aggregate_data[url]
+        cold_time_min = None
+        modified_time_min = None
+        if 'cold_times' in url_results:
+            for trial in url_results['cold_times']:
+                if type(url_results['cold_times'][trial]) == float or \
+                        type(url_results['cold_times'][trial]) == int:
+                            url_results['cold_times'][trial] = \
+                            [url_results['cold_times'][trial]]
+                if cold_time_min is None:
+                    cold_time_min = min(url_results['cold_times'][trial])
+                else:
+                    cold_time_min = min(cold_time_min,
+                            min(url_results['cold_times'][trial]))
+        if 'modified_cold_times' in url_results:
+            for trial in url_results['modified_cold_times']:
+                if type(url_results['modified_cold_times'][trial]) == float or \
+                        type(url_results['modified_cold_times'][trial]) == int:
+                            url_results['modified_cold_times'][trial] = \
+                            [url_results['modified_cold_times'][trial]]
+                if modified_time_min is None:
+                    modified_time_min = \
+                        min(url_results['modified_cold_times'][trial])
+                else:
+                    modified_time_min = min(cold_time_min,
+                            min(url_results['modified_cold_times'][trial]))
+        url_min_results['cold_time'] = cold_time_min
+        url_min_results['modified_cold_time'] = modified_time_min
+        min_data[url] = url_min_results
+
+    pickle.dump(min_data, open('data/results.db', 'wb'))
+
+def generate_hars():
+    """Merge plts and wprs into a har file
+
+    Stores 2 .har files per url in /home/jamshed/page_load_time/data/har/
+    One named with the original url, and one appended with '_modified'
+    """
+    results_path = \
+            '/home/jamshed/page_load_time/telemetry/data/results.db'
```

Comment:
Is it possible for us to use relative urls instead of absolute urls? At least then it's a bit less hardcoded, e.g. I could clone the repo and run it in /home/cs without problems.